lang/attribute/program/src/declare_program/common.rs

```diff
@@ -38,7 +38,10 @@ pub fn gen_accounts_common(idl: &Idl, prefix: &str) -> proc_macro2::TokenStream
 }
 
 pub fn convert_idl_type_to_syn_type(ty: &IdlType) -> syn::Type {
-    syn::parse_str(&convert_idl_type_to_str(ty)).unwrap()
+    match ty {
+        IdlType::Bytes => syn::parse_str("Vec<u8>").unwrap(),
+        _ => syn::parse_str(&convert_idl_type_to_str(ty)).unwrap(),
+    }
```

Comment:
I think this would currently only *partly* fixes the issue since `convert_idl_type_to_str` is being used in multiple places.

We should instead change `bytes` to `Vec<u8>` here: https://github.com/coral-xyz/anchor/blob/460a16171a715671f77ead5629391c0466366c08/lang/attribute/program/src/declare_program/common.rs#L62